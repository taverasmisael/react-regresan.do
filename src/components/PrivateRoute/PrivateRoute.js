import React from 'react'
import { Route, Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'

const IsAuthorized = () => false

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route {...rest} render={props => (IsAuthorized() ? <Component {...props} /> : <Redirect to="/login" />)} />
)

PrivateRoute.propTypes = {
  component: PropTypes.func.isRequired
}

export default PrivateRoute
