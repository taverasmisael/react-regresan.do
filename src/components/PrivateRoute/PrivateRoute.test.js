import React from 'react'
import { Redirect, Route } from 'react-router-dom'

import { shallow } from 'enzyme'

import PrivateRoute from './PrivateRoute'

let canActivate
let cannotActivate
let baseComponent

describe('components/PrivateRoute', () => {
  beforeAll(() => {
    canActivate = jest.fn(() => true)
    cannotActivate = jest.fn(() => false)
    baseComponent = jest.fn()
  })
  describe('rendering', () => {
    it('renders without crashing', () => {
      shallow(<PrivateRoute canActivate={canActivate} component={baseComponent} />)
    })
    it('matches snapshot', () => {
      expect(shallow(<PrivateRoute canActivate={canActivate} component={baseComponent} />)).toMatchSnapshot()
      expect(shallow(<PrivateRoute canActivate={cannotActivate} component={baseComponent} />)).toMatchSnapshot()
      expect(shallow(<PrivateRoute canActivate={canActivate} component={baseComponent} />)).toMatchSnapshot()
    })
  })
  describe('functionality', () => {
    it('redirects if "canActivate" return false', () => {
      const component = shallow(<PrivateRoute canActivate={cannotActivate} component={baseComponent} />)
      expect(component.find(Redirect)).toHaveLength(1)
    })
    it('renders the right route if "canActivate" return true', () => {
      const component = shallow(<PrivateRoute canActivate={canActivate} component={baseComponent} />)
      expect(component.find(Route)).toHaveLength(1)
    })
  })
})
