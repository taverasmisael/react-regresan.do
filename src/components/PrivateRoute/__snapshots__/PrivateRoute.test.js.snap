// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PrivateRoute rendering matches snapshot 1`] = `
<Route
  component={[Function]}
/>
`;

exports[`components/PrivateRoute rendering matches snapshot 2`] = `
<Redirect
  push={false}
  to="/login"
/>
`;

exports[`components/PrivateRoute rendering matches snapshot 3`] = `
<Route
  component={[Function]}
/>
`;
